backend/tests/Feature/SubmissionTest.php

```diff
@@ -549,6 +549,73 @@ public function testReviewersCannotAssignRoles($role)
         $response->assertJsonPath('errors.0.extensions.category', 'authorization');
     }
 
+    /**
+     * @dataProvider allSubmissionRoles
+     * @param string $role
+     * @return void
+     */
+    public function testMyRoleFields(string $role): void
+    {
+        /** @var User $user */
+        $user = User::factory()->create();
+        $this->actingAs($user);
+        $submission = Submission::factory()->create();
+        $camelized = Str::camel($role);
+        $submission->$camelized()->attach($user);
+        $graphql = '
+            query GetSubmission($id: ID!) {
+                submission(id: $id) {
+                    my_role
+                    effective_role
+                }
+            }
+        ';
+
+        $response = $this->graphQL($graphql, ['id' => $submission->id]);
+        $response
+            ->assertJsonPath('data.submission.my_role', Str::singular($role))
+            ->assertJsonPath('data.submission.effective_role', Str::singular($role));
+    }
+
+    public function testAdminGetsEffectiveRole()
+    {
+        $this->beAppAdmin();
+        $submission = Submission::factory()->create();
+        $gql = '
+            query GetSubmission($id: ID!) {
+                submission(id: $id) {
+                    my_role
+                    effective_role
+                }
+            }
+        ';
+
+        $this->graphQL($gql, ['id' => $submission->id])
+            ->assertJsonPath('data.submission.my_role', null)
+            ->assertJsonPath('data.submission.effective_role', 'review_coordinator');
+    }
+
+    public function testPublicationAdminGetsEffectiveRole()
+    {
+        /** @var User $user */
+        $user = User::factory()->create();
+        $this->actingAs($user);
+        $submission = Submission::factory()->create();
+        $submission->publication->editors()->attach($user);
```

Comment:
This user is assigned as an editor, not a publication administrator. I recommend one of the following:

1. Change the name of this test and add a new test case for publication admins
2. Change the role assignment in this test and add a new test case for editors
3. Add a publication roles provider and use that on this test like `allSubmissionRoles` is used elsewhere

Reply:
This legit doesn't matter, but I'll update it.